### ISSUE TITLE
Module LANGUAGE-PARSING not found

### ISSUE THREAD

**Issue by daejunpark** [kframework/k-legacy#1985]
@cos do you have any idea?

```
require "domains.k"

module B-SYNTAX
    syntax Pgm ::= "run"
endmodule

module B
    imports B-SYNTAX

    configuration <T>
        <k> $PGM:Pgm </k> <a> $XA:Int </a>
    </T>

    rule run => 0
endmodule
```

```
$ krun -cXA=1 1.b
[Error] Critical: Parser returned a non-zero exit code: 113
Stdout:

Stderr:
[Error] Inner Parser: Module LANGUAGE-PARSING not found. Specify a module with
-m.
```

```
$ cat 1.b
run
```

**Comment by cos** [comment]
`LANGUAGE-PARSING` is added by the compilation pipeline. Could you --debug this?

**Comment by daejunpark** [comment]
```
$ krun  --debug -cXA=1 1.b
org.kframework.utils.errorsystem.ParseFailedException: [Error] Critical: Parser returned a non-zero exit code: 113
Stdout:

Stderr:
[Error] Inner Parser: Module LANGUAGE-PARSING not found. Specify a module with
-m.

    at org.kframework.krun.KRun.externalParse(KRun.java:317)
    at org.kframework.krun.KRun.parseConfigVars(KRun.java:262)
    at org.kframework.krun.KRun.run(KRun.java:70)
    at org.kframework.krun.KRunFrontEnd.run(KRunFrontEnd.java:91)
    at org.kframework.main.FrontEnd.main(FrontEnd.java:52)
    at org.kframework.main.Main.runApplication(Main.java:110)
    at org.kframework.main.Main.runApplication(Main.java:100)
    at org.kframework.main.Main.main(Main.java:52)
[Error] Critical: Parser returned a non-zero exit code: 113
Stdout:

Stderr:
[Error] Inner Parser: Module LANGUAGE-PARSING not found. Specify a module with
-m.
```

**Comment by daejunpark** [comment]
https://github.com/kframework/k/blob/b422c31f1f10ba4fa028198b9da05eff7b2c4ca3/kernel/src/main/java/org/kframework/krun/KRunOptions.java#L117

Clue: If I drop the option `-m mainModuleName` for kast, then it works.

**Comment by daejunpark** [comment]
BTW, do you know why we have duplicate configVars methods? I think the first one should be removed:
https://github.com/kframework/k/blob/b422c31f1f10ba4fa028198b9da05eff7b2c4ca3/kernel/src/main/java/org/kframework/krun/KRunOptions.java#L96
https://github.com/kframework/k/blob/b422c31f1f10ba4fa028198b9da05eff7b2c4ca3/kernel/src/main/java/org/kframework/krun/KRunOptions.java#L114

**Comment by cos** [comment]
I agree.

**Comment by cos** [comment]
@daejunpark, I don't understand how things are wired in the frontend and options. I can at most make a guess. This line in `KRun`:

```
: options.configurationCreation.configVars(compiledDef.languageParsingModule().name()).entrySet()) {
```

probably shouldn't take `languageParsingModule`. I would guess `compiledDef.parsedDefinition.mainModule` is a better choice. If it doesn't work, ask @dwightguth about it.

**Comment by daejunpark** [comment]
fixed via #1995
